org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServiceAccessor.java

```diff
@@ -204,6 +206,27 @@ private static boolean capabilitiesComply(LanguageServerWrapper wrapper,
 				|| capabilitiesPredicate.test(castNonNull(wrapper.getServerCapabilities()));
 	}
 
+	/**
+	 * Asynchronously checks if the given {@code wrapper}'s capabilities comply with the given
+	 * {@code capabilitiesPredicate}.
+	 *
+	 * @param wrapper
+	 *            the server that's capabilities are tested with
+	 *            {@code capabilitiesPredicate}
+	 * @param capabilitiesPredicate
+	 *            predicate testing the capabilities of {@code wrapper}.
+	 * @return The result of applying the capabilities of {@code wrapper} to
+	 *         {@code capabilitiesPredicate}, or {@code false} if
+	 *         {@code capabilitiesPredicate == null} or
+	 *         {@code wrapper.getServerCapabilities() == null}
+	 */
+	private static CompletableFuture<Boolean> capabilitiesComplyAsync(final LanguageServerWrapper wrapper,
+			final @Nullable Predicate<ServerCapabilities> capabilitiesPredicate) {
+		return capabilitiesPredicate == null //
+				? CompletableFuture.completedFuture(false) //
```

Comment:
Am I misreading the existing method `capabilitiesComply`? Is that method not returning true if the predicate is null? If so, should we not change the documentation to match the implementation and return `CompletableFuture.completedFuture(true)`?

Reply:
Good catch. fixed.